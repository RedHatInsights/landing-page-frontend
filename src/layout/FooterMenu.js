import React from 'react';
import {
    Grid,
    GridItem,
    PageSection,
    TextContent,
    Text,
    TextVariants,
    Split,
    SplitItem
} from '@patternfly/react-core';
import { PaperPlaneAltIcon } from '@patternfly/react-icons';

import './FooterMenu.scss';

const FooterMenu = () => (
    <PageSection className='ins-c-footer__menu pf-m-no-fill'>
        <Grid sm={ 12 } md={ 3 } gutter="lg">
            <GridItem>
                <TextContent>
                    <Text component={ TextVariants.h2 }>Support and resources</Text>
                </TextContent>
                <a href="https://access.redhat.com/support">Get support</a>
                <a href="https://access.redhat.com/support/customer-service">Contact customer service</a>
            </GridItem>
            <GridItem>
                <TextContent>
                    <Text component={ TextVariants.h2 }>About</Text>
                </TextContent>
                <a href="https://www.redhat.com/en/technologies/management/insights">Red Hat Insights</a>
                <a href="https://www.redhat.com/en/technologies/management/smart-management">Red Hat Smart Management</a>
                { /*
                    TODO: Unhide at a later point
                    <a href="#">Red Hat Hybrid CloudManagement</a>
                */ }
                <a href="https://www.redhat.com/en/technologies/cloud-computing/openshift">Red Hat OpenShift</a>
            </GridItem>
            <GridItem>
                <TextContent>
                    <Text component={ TextVariants.h2 }>Feedback</Text>
                </TextContent>
                <Split gutter="md">
                    <SplitItem><PaperPlaneAltIcon size='xl' className="ins-c-icon__active"/></SplitItem>
                    <SplitItem isFilled><a className='ins-m-inline'>Tell us about your experience</a> using
                    Red Hat Cloud Services software, and how we can improve.</SplitItem>
                </Split>
            </GridItem>
        </Grid>
    </PageSection>
);

export default FooterMenu;
