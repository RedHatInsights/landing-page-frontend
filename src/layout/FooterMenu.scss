@import '~@red-hat-insights/insights-frontend-components/Utilities/_mixins';

.ins-c-footer__menu{
    background: white;
    box-shadow: var(--pf-global--BoxShadow--sm);
    z-index: 2;
    @include rem('padding', 60px);
    a {
        display: block;
        & + a { @include rem('margin-top', 15px); }
    }
}