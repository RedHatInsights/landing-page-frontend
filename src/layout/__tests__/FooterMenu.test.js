import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FooterMenu from '../FooterMenu';

describe('Footer Menu component', () => {
    it('should render correctly', () => {
        const wrapper = shallow(<FooterMenu />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
