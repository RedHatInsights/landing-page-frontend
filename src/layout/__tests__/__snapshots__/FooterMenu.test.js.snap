// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Menu component should render correctly 1`] = `
<PageSection
  className="ins-c-footer__menu pf-m-no-fill"
>
  <Grid
    gutter="lg"
    md={3}
    sm={12}
  >
    <GridItem>
      <TextContent>
        <Text
          component="h2"
        >
          Support and resources
        </Text>
      </TextContent>
      <a
        href="https://access.redhat.com/support"
      >
        Get support
      </a>
      <a
        href="https://access.redhat.com/support/customer-service"
      >
        Contact customer service
      </a>
    </GridItem>
    <GridItem>
      <TextContent>
        <Text
          component="h2"
        >
          About
        </Text>
      </TextContent>
      <a
        href="https://access.redhat.com/products/red-hat-insights#getstarted"
      >
        Red Hat Insights
      </a>
      <a
        href="https://access.redhat.com/products/cloud_management_services_for_rhel"
      >
        Red Hat Smart Management
      </a>
      <a
        href="https://www.openshift.com/"
      >
        Red Hat OpenShift
      </a>
    </GridItem>
    <GridItem>
      <TextContent>
        <Text
          component="h2"
        >
          Feedback
        </Text>
      </TextContent>
      <Split
        gutter="md"
      >
        <SplitItem>
          <PaperPlaneAltIcon
            className="ins-c-icon__active"
            color="currentColor"
            noVerticalAlign={false}
            size="xl"
            title={null}
          />
        </SplitItem>
        <SplitItem
          isFilled={true}
        >
          <a
            className="ins-m-inline"
            href="javascript:;"
            tabIndex="0"
          >
            Tell us about your experience
          </a>
           using Red Hat Cloud Services software, and how we can improve.
        </SplitItem>
      </Split>
    </GridItem>
  </Grid>
</PageSection>
`;
